lib/buffered_logger/log_device_proxy.rb

```diff
@@ -12,44 +14,54 @@ def close
     end
 
     def end
-      @logdev.write(@buffers.delete(key).string)
+      @logdev.write(string_io.string)
+      destroy_thread_local
     end
 
     def flush
-      log, @buffers[key] = @buffers.delete(key).string, StringIO.new
-      @logdev.write(log)
+      @logdev.write(string_io.string)
+      string_io.string.clear
     end
 
     def start
-      @buffers[key] = StringIO.new
+      self.string_io = StringIO.new
     end
 
     def started?
-      @buffers.key?(key)
-    end
-
-    def sweep
-      @buffers.clone.each do |key, buffer|
-        @buffers.delete(key) unless key.all?(&:alive?)
-      end
-      true
+      !!string_io
     end
 
     def write(message)
       if started?
-        @buffers[key].write(message)
+        string_io.write(message)
       else
         @logdev.write(message)
       end
     end
 
     def current_log
-      @buffers[key].string.dup
+      string_io.string.dup
     end
 
     private
-      def key
-        [Thread.current]
+    def string_io
+      if Thread.current.respond_to?(:thread_variable_get)
+        Thread.current.thread_variable_get(THREAD_LOCAL_VAR_NAME)
```

Comment:
Makes sense. A while back I had the key based on the current Thread and Fiber but realized it didn't make any sense to include the Fiber because Fibers are concurrent. `#thread_variable_get` seems like a good idea, maybe we should make the gem require ruby 2.0 and above and just ditch using `#[]`.